generators/gulp/templates/gulp/config.js

```diff
@@ -1,22 +1,24 @@
+<% isCraft = platform == 'craft' -%>
+<% var rootSrcPath = isCraft ? 'public' : 'dist' -%>
 //
 //   Config
 //
 //////////////////////////////////////////////////////////////////////
 
 var paths = {
-  dist: 'dist/',
+  dist: '<%= rootSrcPath %>/',
```

Comment:
Setting this to `public` doesn't work so well. It means that everything in `public` will get blown away with gulp. We should set it to `public/dist` for Craft installs and `dist` for non-craft.